compiler/rustc_codegen_llvm/src/back/write.rs

```diff
@@ -1042,7 +1043,7 @@ unsafe fn embed_bitcode(
             llvm::LLVMSetInitializer(llglobal, llconst);
 
             let section = bitcode_section_name(cgcx);
-            llvm::LLVMSetSection(llglobal, section.as_ptr().cast());
+            llvm::LLVMSetSection(llglobal, section.as_c_char_ptr());
```

Comment:
`LLVMSetSection` accepts `const char*` without length, where `section` is &str.

Ahh, `bitcode_section_name` have fixme, so it can actually return cstr literals.

Reply:
Yeah, and all of the calls to `llvm::LLVMSetSection` should really be calling the safe wrapper `llvm::set_section` (which takes `&CStr`), which would enforce type-safety.

Reply:
https://github.com/rust-lang/rust/pull/132340